gradle-jdks-setup/README.md

```diff
@@ -0,0 +1,246 @@
+Note that now, when running any `./gradlew` command the first log lines would say that:
+* a toolchain is installed already:
+```
+Distribution 'https://corretto.....' already exists in '/.gradle/gradle-jdks/amazon-corretto-11.0.23.9.1-d6ef2c62dc4d4dd4'
+```
+* or that a toolchain needs to be installed:
+```
+JDK installation '/.gradle/gradle-jdks/amazon-corretto-11.0.23.9.1-d6ef2c62dc4d4dd4' does not exist, installing 'https://corretto.....' in progress ...
+```
+
+7. Check the gradle version
+```
+./gradlew --version
+```
+
+The JVM should correspond to the java major version configured in step 3.
+
+
+## Gradle JDK Configuration directory structure
+
+The Gradle JDK configuration will be rendered **automatically** by `./gradlew generateGradleJdkConfigs` in a directory format structure as it needs to be easily parsed both from Bash and from Java (see [Entry points](#how-it-works-))
+
+DO NOT MANUALLY CHANGE THE FILES, it may break the JDK setup workflow. Instead, please update the [JdksExtension](../gradle-jdks/src/main/java/com/palantir/gradle/jdks/JdksExtension.java) or update the [`gradle-jdks-latest`](https://github.com/palantir/gradle-jdks-latest) version.
+```
+project-root/
+├── gradle/
+│   ├── wrapper/
+│   │   ├── gradle-wrapper.jar
+│   │   ├── gradle-wrapper.properties
+│   ├── jdks/
+│   │   ├── <jdkMajorVersion eg.11>/
+│   │   │   ├── <os eg. linux>/
+│   │   │   │   ├── <arch eg. aarch64>/
+│   │   │   │   │   ├── download-url
+│   │   │   │   │   ├── local-path
+│   ├── certs/
+│   │   ├── Palantir3rdGenRootCa.serial-number
+│   ├── gradle-daemon-jdk-version
+│   ├── gradle-jdks-setup.sh
+│   ├── gradle-jdks-setup.jar
+├── subProjects/...
+```
+
+- `gradle/gradle-daemon-jdk-version` 
+  - contains the Gradle JDK Daemon version rendered from `JdksExtension#daemonTarget`
+- `gradle/certs/` 
+  - contains a list of certificates (as serial numbers) that need to be added to the configured JDKs in `gradle/jdks/*`.
+  - is rendered from `JdksExtension#caCerts`
+- `gradle/jdks`
+  - contains a list of directories in the format `<jdk_major_version>/<os>/<arch>` that contain 2 files: 
+    - `download-url` full url path for the jdk, os and arch. Rendered from `JdksExtension#jdks` configured in step 2
+    - `local-path` the local name of the file. Rendered based on the distribution-name, version and the [hash](../gradle-jdks/src/main/java/com/palantir/gradle/jdks/JdkSpec.java) 
+  - it generates all the JDK versions configured in [JdksExtension](../gradle-jdks/src/main/java/com/palantir/gradle/jdks/JdksExtension.java)
+
+Running the patched `./gradlew` script will add extra configurations required for Intelij:
+```
+project-root/
+├── .gradle/
+│   ├── config.properties
+├── .idea/
+│   ├── runConfigurations/
+│   │   ├── IntelijGradleJdkSetup.xml
```

Comment:
This is currently not gitignored and should be (if you run `git clean -xn .idea` you'll see what would not exist in a clean checkout)

Reply:
yes! I have a fix for this on the way